src/renderer/components/AccountView.tsx

```diff
@@ -37,28 +38,64 @@ const explorerURL = (net: Net, address: string) => {
   }
 };
 
-function AccountView(props: { pubKey: string | undefined }) {
+function AccountView(props: { pubKey: string }) {
   const { pubKey } = props;
   const { net } = useAppSelector(selectValidatorNetworkState);
 
-  const [account, setSelectedAccountInfo] = useState<AccountInfo | undefined>(
-    undefined
-  );
+  if (!pubKey) {
+    pubKey = '12341234123412341234';
+  }
 
-  useInterval(() => {
+  const [account, setSelectedAccountInfo] = useState<AccountInfo>({
+    net,
+    pubKey,
+    accountId: new sol.PublicKey(pubKey),
+    accountInfo: sol.AccountInfo < Buffer > {},
+    solDelta: 0,
+    count: 0,
+    maxDelta: 0,
+    programID: '',
+  });
+
+  const updateAccount = useCallback(() => {
+    let ok = false;
     if (pubKey) {
       getAccount(net, pubKey)
-        .then((a) => setSelectedAccountInfo(a))
+        .then((res) => {
+          // eslint-disable-next-line promise/always-return
+          if (res) {
+            setSelectedAccountInfo(res);
+            ok = true;
+          }
+        })
         /* eslint-disable no-console */
         .catch(console.log);
-    } else {
-      setSelectedAccountInfo(undefined);
     }
-  }, 666);
+    if (!ok) {
```

Comment:
I don't think this whole `ok` thing you're looking to do will work the way you intend. The code isn't synchronous, so you can't set results like that -- the `if !(ok)` check will happen before `getAccount` promise returns and calls `.then`. (At least that's what ended up fixing the flickering I was seeing, taking out the `!(ok)` bit)

Reply:
Why not re-write the section(s) like this? 

```typescript
const updateAccount = useCallback(() => {
  async function fetchAccountData() {
    if (pubKey) {
      const res = await getAccount(net, pubKey);
      if (res) {
        setChangeInfo(res);
      } else {
        const offChainAccount: AccountInfo = {
          net,
          pubKey,
          accountId: new sol.PublicKey(pubKey),
          accountInfo: {
            executable: false,
            lamports: 0,
            owner: sol.SystemProgram.programId,
            data: Buffer.from(''),
          },
          solDelta: 0,
          count: 0,
          maxDelta: 0,
          programID: '',
        };
        setChangeInfo(offChainAccount);
      }
    }
  }
  fetchAccountData();
}, [net, pubKey]);
useEffect(updateAccount, [net, pubKey, updateAccount]);
useInterval(updateAccount, 666);
```

If you define the `async` function within the callback/`useEffect`, you can use `await`.

Reply:
Why not re-write to something like this?

```typescript
const updateAccount = useCallback(() => {
  async function fetchAccountData() {
    if (pubKey) {
      const res = await getAccount(net, pubKey);
      if (res) {
        setChangeInfo(res);
      } else {
        const offChainAccount: AccountInfo = {
          net,
          pubKey,
          accountId: new sol.PublicKey(pubKey),
          accountInfo: {
            executable: false,
            lamports: 0,
            owner: sol.SystemProgram.programId,
            data: Buffer.from(''),
          },
          solDelta: 0,
          count: 0,
          maxDelta: 0,
          programID: '',
        };
        setChangeInfo(offChainAccount);
      }
    }
  }
  fetchAccountData();
}, [net, pubKey]);
useEffect(updateAccount, [net, pubKey, updateAccount]);
useInterval(updateAccount, 666);
```

Reply:
Ha! you're right, this programming paradigm is such a joy to internalise